lib/error/ApiAuthRequestError.js

```diff
@@ -2,12 +2,15 @@
 
 var utils  = require('../utils');
 
-function ApiAuthRequestError(message, statusCode){
+function ApiAuthRequestError(errorData){
   Error.call(this);
   Error.captureStackTrace(this, this.constructor);
   this.name='ApiAuthRequestError';
-  this.userMessage = this.message = message;
-  this.statusCode = statusCode || 400;
+  this.userMessage = this.message = errorData.userMessage;
+  this.statusCode = errorData.statusCode || 400;
+  if(errorData.error){
```

Comment:
Instead of the `if`, how about:

```javascript
this.error = errorData.error || undefined;
```

Or if `null` and other falsey values are okay to pass on, just assign is directly (without the `if`):

```javascript
this.error = errorData.error;
```

What do you think?